src/main/java/com/marklogic/appdeployer/command/ResourceManagerImpl.java

```diff
@@ -0,0 +1,161 @@
+		initialize();
+	}
+
+	@Override
+	public void initialize() {
+		File propertiesFile = new File(hashFilePath);
+		logger.info("Loading properties from: " + propertiesFile.getAbsolutePath());
+		propertiesFile.getParentFile().mkdirs();
+		if (propertiesFile.exists()) {
+			FileInputStream fis = null;
+			try {
+				fis = new FileInputStream(propertiesFile);
+				if (logger.isDebugEnabled()) {
+					logger.debug("Loading properties from: " + propertiesFile.getAbsolutePath());
+				}
+				props.load(fis);
+			} catch (Exception e) {
+				logger.warn("Unable to load properties, cause: " + e.getMessage());
+			} finally {
+				try {
+					fis.close();
+				} catch (Exception e) {
+					logger.warn(e.getMessage());
+				}
+			}
+		}
+	}
+
+	@Override
+	public boolean hasFileBeenModifiedSinceLastDeployed(File file) {
+		String lastDeployedChecksum = null;
+		try {
+			lastDeployedChecksum = computeFileChecksum(md5Digest, file);
+		} catch (IOException ie) {
+			throw new RuntimeException("Unable to compute hash on file from path: " + file.getAbsolutePath() + "; cause: " + ie.getMessage(), ie);
+		}
+		if (lastDeployedChecksum != null) {
+			String key = buildKey(file);
+			String lastChecksum = props.getProperty(key);
+			if (logger.isDebugEnabled()) {
+				logger.debug("lastDeployedChecksum: " + lastDeployedChecksum);
+				logger.debug("key: " + key);
+				logger.debug("lastChecksum: " + lastChecksum);
+			}
+			if (lastChecksum != null) {
+				return !lastChecksum.equals(lastDeployedChecksum);
+			} else {
+				return true;
+			}
+		} else {
+			logger.info("Hmm, really shouldn't get here. lastDeployedChecksum should never be null here. Should have thrown an exception if there was a problem. Probably need to refactor this out.");
+			return false;
+		}
+	}
+
+	@Override
+	public void saveLastDeployedHash(File file) {
+		String key = buildKey(file);
+		String fileChecksum = null;
+		try {
+			fileChecksum = computeFileChecksum(md5Digest, file);
+		} catch (IOException ie) {
+			throw new RuntimeException("Unable to compute hash on file from path: " + file.getAbsolutePath() + "; cause: " + ie.getMessage(), ie);
+		}
+		props.setProperty(key, fileChecksum);
+		FileWriter fw = null;
+		try {
+			fw = new FileWriter(new File(hashFilePath));
+			props.store(fw, "");
+		} catch (Exception e) {
+			logger.warn("Unable to store properties, cause: " + e.getMessage());
+		} finally {
+			try {
+				fw.close();
+			} catch (Exception e) {
+				logger.warn(e.getMessage());
+			}
+		}
+	}
+
+	private String computeFileChecksum(MessageDigest digest, File file) throws IOException {
+		//Get file input stream for reading the file content
+		FileInputStream fis = new FileInputStream(file);
+
+		//Create byte array to read data in chunks
+		byte[] byteArray = new byte[1024];
+		int bytesCount = 0;
+
+		//Read file data and update in message digest
+		while ((bytesCount = fis.read(byteArray)) != -1) {
+			digest.update(byteArray, 0, bytesCount);
```

Comment:
If the static block above fails, then digest will be null. I don't think we ever expect the static block to fail, but if digest is null, then I think this class should immediately check that and bail with a log warning message (bail = don't do anything).